pkg/minishift/systemtray/systemtray.go

```diff
@@ -0,0 +1,310 @@
+		startMenu := submenu.AddSubMenuItem(START, "", 0)
+		stopMenu := submenu.AddSubMenuItem(STOP, "", 0)
+		submenus[profile] = submenu
+		submenusToMenuItems[profile] = MenuAction{start: startMenu, stop: stopMenu}
+	}
+
+	go func() {
+		<-exit.OnClickCh()
+		systray.Quit()
+	}()
+
+	for k, v := range submenusToMenuItems {
+		go startStopHandler(icon.Running, k, v.start, START_PROFILE)
+		go startStopHandler(icon.Stopped, k, v.stop, STOP_PROFILE)
+	}
+
+	go addNewProfilesToTray()
+
+	go removeDeletedProfilesFromTray()
+
+	go updateProfileStatus()
+}
+
+func OnExit() {
+	return
+}
+
+func getStatus(profileName string) int {
+	api := libmachine.NewClient(state.InstanceDirs.Home, state.InstanceDirs.Certs)
+	defer api.Close()
+
+	vmStatus, err := cluster.GetHostStatus(api, profileName)
+	if err != nil {
+		return DOES_NOT_EXIST
+	}
+
+	if vmStatus == libmachineState.Running.String() {
+		return RUNNING
+	}
+	if vmStatus == libmachineState.Stopped.String() {
+		return STOPPED
+	}
+	return 0
+}
+
+// Add newly created profiles to the tray
+func addNewProfilesToTray() {
+	for {
+		time.Sleep(40 * time.Second)
+
+		newProfilesList := profile.GetProfileList()
+		for _, profile := range newProfilesList {
+			submenusLock.Lock()
+			if _, ok := submenus[profile]; ok {
+				submenusLock.Unlock()
+				continue
+			} else {
+				submenu := systray.AddSubMenu(strings.Title(profile))
+				submenus[profile] = submenu
+				submenusLock.Unlock()
+				startMenu := submenu.AddSubMenuItem(START, "", 0)
+				stopMenu := submenu.AddSubMenuItem(STOP, "", 0)
+				submenusToMenuItemsLock.Lock()
+				ma := MenuAction{start: startMenu, stop: stopMenu}
+				submenusToMenuItems[profile] = ma
+				submenusToMenuItemsLock.Unlock()
+
+				go startStopHandler(icon.Running, profile, ma.start, START_PROFILE)
+
+				go startStopHandler(icon.Stopped, profile, ma.stop, STOP_PROFILE)
+			}
+		}
+	}
+}
+
+// Remove deleted profiles from tray
+func removeDeletedProfilesFromTray() {
+	for {
+		time.Sleep(30 * time.Second)
+		newProfileList := profile.GetProfileList()
+		for k := range submenus {
+			submenusLock.Lock()
+			if exists, _ := slice.ItemExists(newProfileList, k); exists {
+				submenusLock.Unlock()
+				continue
+			} else {
+				submenus[k].Hide()
+				delete(submenus, k)
+				submenusLock.Unlock()
+				if _, ok := submenusToMenuItems[k]; ok {
+					submenusToMenuItemsLock.Lock()
+					delete(submenusToMenuItems, k)
+					submenusToMenuItemsLock.Unlock()
+				}
+			}
+		}
+	}
+}
+
+// stopProfile stops a profile when clicked on the stop menuItem
+func stopProfile(profileName string) error {
+	minishiftBinary, _ := os.CurrentExecutable()
+
+	if runtime.GOOS == "windows" {
+		cmd, err := exec.LookPath("cmd.exe")
+		if err != nil {
+			if glog.V(3) {
+				fmt.Println("Could not find cmd.exe in path")
+				return fmt.Errorf("%v", err)
+			}
+		}
+		args := []string{"/C", "start", minishiftBinary, "stop", "--profile", profileName}
+		command := exec.Command(cmd, args...)
```

Comment:
@gbraad This is the code that runs the `minishift start --profile <profileName>` command in `cmd.exe`, maybe i am doing something wrong here? which is the reason why we are seeing this https://github.com/minishift/minishift/pull/2633#issuecomment-425339621